docs/experiment/guides/troubleshooting/variant-jumping.md

```diff
@@ -0,0 +1,92 @@
+
+    You may find it helpful to "only show" assignment and exposure events in the user timeline.
+
+Important things to note when debugging a user timeline include:
+
+1. Did you introduce any [targeting changes](#targeting-changes) while your flag/experiment was active? Could the timing of this change have effected the variant assigned to this user?
+2. What is the bucketing key used in the flag/experiment? Does the value for this property change between assignments/exposures for this user?
+3. Does the user seem have "missing" exposures or assignment events? If so the "missing" event may have been sent for a "different" user.
+4. Does an assignment not have a user property that is expected? If so, double check the server upload timestamp of the assignment event vs surrounding active events. You may notice that events sent from the client are actually uploaded and received after the assignment event, even if the client order is different.
+
+## Normal variant jumping
+
+Normal variant jumping may occur due to:
+
+1. [Targeting changes](#targeting-changes): changes to targeting rules while your experiment is running
+2. [Anonymous identity merging](#anonymous-identity-merging): anonymous users, bucketed by amplitude ID may receive different variants until they are eventually resolved via a matching user ID.
+
+### Targeting changes
+
+Performing the following actions may cause a user to jump variants:
+
+1. Adding or removing a variant
+2. Changing variant distribution weights
+3. Targeting dynamic cohorts
+4. Changing the bucketing key
+5. Updating mutual exclusion
+
+!!!tip "Avoid variant jumping by enabling sticky bucketing"
+    Enabling sticky bucketing before making targeting changes will avoid variant jumping. However, sticky bucketing may cause a [sample ratio mismatch (SRM)](./sample-ratio-mismatch.md).
+
+### Anonymous identity merging
+
+Variant jumping caused by anonymous identity merging may occur due to bucketing by Amplitude ID, when user profiles with distinct device IDs (resolved to different amplitude IDs) are merged when resolved to a common user ID. This may happen if a user uses your app on different devices, or if, on logout, the device ID is re-generated.
+
+!!!info "[Learn more about Amplitude's identity resolution and merging users.](https://help.amplitude.com/hc/en-us/articles/115003135607-Track-unique-users-in-Amplitude#h_e9913ce1-549a-4d88-a832-1f56aede581a)"
+
+ Identifying this type of variant jumping is easy by identifying the the assignment event where the user jumped between variants, and comparing the value of the Amplitude ID for both events. If the Amplitude ID is different on the two events, then there then it is very likely that anonymous identity merging was the cause.
+
+ To combat this type of variant jumping, consider bucketing by user ID. Note that bucking by user ID may not be possible if you expect that users in the experiment will be anonymous.
```

Comment:
If they have anonymou users, is our suggestion to bucket by device id

Reply:
Hmmm, actually targeting by device ID may very likely increase variant jumping. Since when taking unique users in amplitude we consider users by amplitude ID, not device ID.

Reply:
I guess if you're certain the the user is anonymous then its ok